# Task 03 - Business Intelligence Dashboard

### 3. Business Understanding

### 3.1 Problem Statement

The dataset contains transactional sales data that captures the relationship between customers, products, and purchasing activities. The primary challenge is to understand how customer characteristics, product categories, locations, and purchasing patterns influence overall sales performance. By analyzing these interactions, the business can identify meaningful patterns, monitor sales performance, and support data-driven decision-making for future business strategies.

### 3.2 About the Dataset
This dataset represents retail sales transactions involving customers, products, and orders. It captures how different customers purchase different products over time and records the resulting sales generated from those transactions.

The dataset combines customer information, product details, order information, sales values, and geographical information, making it suitable for business intelligence, customer analytics, sales analysis, and dashboard development.

#### Unit of Analysis

The unit of analysis in this dataset is a transaction line item.

Each row represents a single product purchased as part of a customer order rather than an entire customer or a complete order.

A single Order ID may appear multiple times because one order can contain multiple products. Therefore, Order ID should be treated differently from the number of rows while performing business analysis.

#### Main Business Entities

The dataset mainly consists of three business entities:

• Customer
    - Identified using Customer ID
    - Contains demographic information such as Age, Gender and City.

• Order
    - Identified using Order ID and Order Date.
    - Represents purchasing activity performed by customers.

• Product
    - Includes Product Name, Category and Sales information.
    - Represents items purchased during customer transactions.

#### Types of Analysis Supported

This dataset supports multiple business analyses, including:

- Customer Purchasing Behaviour Analysis
- Customer Segmentation Analysis
- Product and Category Performance Analysis
- Time-based Sales Trend Analysis
- Geographical Sales Performance Analysis

#### Dataset Limitations

The dataset does not contain some important business information, including:

- Profit or Cost
- Store or Branch information
- Online/Offline purchase channel
- Customer Reviews or Ratings
- Referral Source
- Marketing Campaign data
- Customer Satisfaction or Returns

Therefore, profitability analysis, customer satisfaction analysis, and marketing effectiveness analysis cannot be performed directly using this dataset.

### 3.3 Business Objectives

1. Understand customer purchasing behavior across different customer segments.

2. Evaluate sales performance across products, categories, cities, and time periods.

3. Identify business opportunities and high-performing areas using sales data.

4. Support data-driven business decision-making through analytical insights and interactive dashboards.

### 3.4 Stakeholders

The primary stakeholders for this analysis are:

#### 1. Business Owner / CEO
- Monitor overall business performance.
- Track sales growth and customer purchasing trends.
- Make strategic business decisions based on business insights.

#### 2. Sales Manager
- Monitor sales performance across products and categories.
- Identify high-performing and low-performing products.
- Improve sales strategies using performance trends.

#### 3. Marketing Manager
- Identify target customer segments.
- Design customer-focused marketing campaigns.
- Promote products based on customer purchasing behaviour and seasonal demand.




### 3.5 Business Questions

#### Customer Purchasing Behaviour

- Which age group generates the highest sales?
- Which age group has the highest average sales per customer?
- Which gender contributes more to overall sales?
- Do loyal customers contribute significantly to total revenue?
- What is the average order value for each customer segment?

---

#### Product Performance

- Which products generate the highest sales?
- Which product categories perform best?
- Are a few products dominating sales, or is demand evenly distributed?
- Which products perform consistently over time?

---

#### Sales Performance

- Which months generate the highest sales?
- Which cities contribute the most revenue?
- Are sales trends consistent throughout the year?
- Which customer segments contribute most during peak seasons?

### 4.1 What is a KPI?

A Key Performance Indicator (KPI) is a measurable business indicator used to evaluate how effectively an organization is achieving a specific business objective.

Unlike general metrics, KPIs are directly linked to business goals and help management monitor performance, identify problems, and make informed business decisions.

In this project, KPIs will be used to measure customer purchasing behaviour, product performance, and overall sales performance.

### 4.2 KPI vs Metric

| Metric | KPI |
|---------|-----|
| Measures any business activity. | Measures progress toward a specific business objective. |
| May or may not be important for decision-making. | Directly supports business decision-making. |
| Used for monitoring general performance. | Used for evaluating strategic business performance. |
| Example: Total Products | Example: Average Order Value |

### 4.3 Characteristics of a Good KPI

A good KPI should have the following characteristics:

- Directly aligned with a business objective.
- Measurable using available data.
- Actionable for business decision-making.
- Trackable over time to monitor trends.
- Easy to understand and communicate.

### 4.4 KPI Selection Framework

Each KPI selected for this project will be evaluated using the following framework:

1. Business Objective
   - Which business objective does the KPI support?

2. Business Rationale
   - Why is this KPI important for the business?

3. Formula
   - How is the KPI calculated?

4. Business Interpretation
   - What does a high or low value indicate?

5. Business Action
   - What decisions or actions can management take based on this KPI?

6. Limitations
   - What information does this KPI not provide?

### 4.5 Final Selected KPIs

Based on the defined business objectives and KPI selection framework, the following KPIs were selected for the dashboard:

1. Total Sales
2. Total Orders
3. Average Order Value (AOV)
4. Repeat Purchase Rate
5. Average Sales per Customer

These KPIs collectively measure overall sales performance, customer purchasing behaviour, and customer loyalty while supporting executive-level business decision-making.

## 5. KPI Implementation & Analysis

### 5.1 Environment Setup

The required Python libraries are imported to perform data manipulation, analysis, and visualization throughout this project.

In [1]:
# Data Manipulation
import pandas as pd
import numpy as np

# Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Notebook Settings
plt.style.use("default")
pd.set_option("display.max_columns", None)

print("Libraries imported successfully.")

Libraries imported successfully.


### 5.2 Load Dataset

The cleaned dataset prepared during Task 2 is loaded for KPI calculation, business analysis, and dashboard development.

In [6]:
# Load Dataset
df = pd.read_csv("../../task 02 - EDA and business analytics/cleaned_ecommerce_data.csv")

print("Dataset loaded successfully.")

Dataset loaded successfully.


### 5.3 Quick Dataset Validation

Before calculating KPIs, a quick validation is performed to ensure that the dataset has been loaded correctly and contains the required information for analysis.

In [7]:
# Shape of dataset
print("Dataset Shape:", df.shape)

# Column Names
print("\nColumns:")
print(df.columns.tolist())

# Missing Values
print("\nMissing Values:")
print(df.isnull().sum())

# Data Types
print("\nData Types:")
print(df.dtypes)

Dataset Shape: (1000, 15)

Columns:
['Order_ID', 'Order_Date', 'Customer_ID', 'Customer_Name', 'Age', 'Gender', 'City', 'Product', 'Category', 'Quantity', 'Unit_Price', 'Total_Sales', 'Calculated_Sales', 'Month', 'Age_Group']

Missing Values:
Order_ID            0
Order_Date          0
Customer_ID         0
Customer_Name       0
Age                 0
Gender              0
City                0
Product             0
Category            0
Quantity            0
Unit_Price          0
Total_Sales         0
Calculated_Sales    0
Month               0
Age_Group           0
dtype: int64

Data Types:
Order_ID                str
Order_Date              str
Customer_ID             str
Customer_Name           str
Age                 float64
Gender                  str
City                    str
Product                 str
Category                str
Quantity              int64
Unit_Price          float64
Total_Sales         float64
Calculated_Sales    float64
Month                   str
Age_Group

### 5.1 KPI 1 - Total Sales

#### Business Purpose

Total Sales represents the overall revenue generated from all recorded transactions. It provides an executive-level summary of business performance and serves as the primary KPI for evaluating revenue generation.

#### Formula

**Total Sales = Sum of Total_Sales**

In [8]:
# KPI 1 : Total Sales

total_sales = df["Total_Sales"].sum()

print(f"Total Sales : ₹{total_sales:,.2f}")

Total Sales : ₹139,399,439.65


#### Business Interpretation

The dataset generated a total revenue of ₹139,399,439.65 across all recorded transactions.

However, this value alone is insufficient for business decision-making. Total Sales becomes meaningful only when compared with previous time periods, predefined business targets, seasonal trends, or industry benchmarks.

Therefore, this KPI serves as a high-level executive indicator and should be analyzed together with trend-based KPIs such as Monthly Sales Trend, Average Order Value, Total Orders, and Customer Growth.

#### Limitations

- Total Sales does not indicate whether business performance is improving or declining.
- It provides no information about profitability.
- It does not explain which products, customers, or regions contributed to the revenue.
- Trend analysis is required before making strategic decisions.

### 5.5 KPI 2 - Total Orders

### 5.6 KPI 3 - Average Order Value (AOV)

### 5.7 KPI 4 - Repeat Purchase Rate

### 5.8 KPI 5 - Average Sales per Customer